mainwindow.cpp

```diff
@@ -1147,8 +1135,16 @@ void MainWindow::setAvailabilityOfFeatures(bool isJumbo)
         QStringList defaultFormats;
         defaultFormats << tr("Auto detect") << "descrypt" << "bsdicrypt" << "md5crypt"
                        << "bcrypt" << "AFS" << "LM" << "crypt" << "tripcode" << "dummy";
+        QString savedFormat = m_ui->comboBox_Format->currentText();
         m_ui->comboBox_Format->clear();
         m_ui->comboBox_Format->addItems(defaultFormats);
+        // Restore user's selection
+        int indexSavedFormat = m_ui->comboBox_Format->findText(savedFormat);
+        if (indexSavedFormat != -1) {
+            m_ui->comboBox_Format->setCurrentIndex(indexSavedFormat);
+        } else if (!savedFormat.isEmpty()){
+            m_ui->comboBox_Format->setEditText(savedFormat);
```

Comment:
Do you even need to do this, isn't that exact text already grabbed from the comboBox? Unless you should have gotten it from settings or something, it look superfluous.